src/compiler.ts

```diff
@@ -0,0 +1,130 @@
+import * as yaml from "js-yaml";
+import * as marked from "marked";
+
+import { StringOrTag } from "./client";
+
+/**
+ * Matches the triple-dash metadata block on the first line of markdown file.
+ * The first capture group contains YAML content.
+ */
+const METADATA_REGEX = /^---\n?((?:.|\n)*)\n---\n/;
+
+/**
+ * Splits text content for lines that begin with `@tagName`.
+ */
+const TAG_REGEX = /^@(\S+)(?:\s+([^\n]+))?$/;
+const TAG_SPLIT_REGEX = /^(@\S+(?:\s+[^\n]+)?)$/gm;
+
+/**
+ * Ignored `@tag` names. Some languages already support `@tags`, so to separate
+ * Documentalist tags, we use these default reserved words to avoid conflicts.
+ *
+ * Plugins may define their own reserved words when calling the `renderBlock`
+ * method.
+ */
+const RESERVED_WORDS = [
+    "import",
+];
+
+/**
+ * The output of `renderBlock` which parses a long form documentation block into
+ * metadata, rendered markdown, and tags.
+ */
+export interface IBlock {
+    /**
+     * The original string content block.
+     */
+    content: string;
+
+    /**
+     * Parsed YAML front matter (if any) or {}.
+     */
+    metadata: any;
+
+    /**
+     * An array of markdown-rendered HTML or tags.
+     */
+    renderedContent: StringOrTag[];
+}
+
+export interface ICompiler {
```

Comment:
wondering if this interface and `IBlock` above should be defined in `plugin.ts`? cuz every plugin will want access to these interfaces so makes sense for them to come from one place.